src/components/dropdown-menu/DropdownMenu.tsx

```diff
@@ -14,18 +14,44 @@ interface DropdownMenuItemProps extends React.HTMLProps<HTMLDivElement> {
   children?: React.ReactNode;
   className?: string;
   onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+  /** To select a certain option in the menu without having to use document.querySelector */
+  forwardRef?: React.RefObject<HTMLDivElement>;
+  onActivate?: (event: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>) => void;
 }
 
 export const DropdownMenuDivider: React.FC = () => <div className="tk-dropdown-menu-divider"></div>
 
-export const DropdownMenuItem: React.FC<DropdownMenuItemProps> = ({children, className, onClick, ...rest}: DropdownMenuItemProps) => {
+export const DropdownMenuItem: React.FC<DropdownMenuItemProps> = ({children, className, onClick, forwardRef, onActivate, ...rest}: DropdownMenuItemProps) => {
   const classes = classNames(
     'tk-dropdown-menu__item',
     className,
   )
 
+  const focusNextOption = (current: HTMLDivElement, direction: number) => {
+    const options = (Array.from(current.parentElement.getElementsByClassName('tk-dropdown-menu__item'))) as HTMLDivElement[];
+    let currentOptionPosition = options.indexOf(current);
+    currentOptionPosition = currentOptionPosition > -1 ? currentOptionPosition : 0;
+    const nextElementPosition = (currentOptionPosition + options.length + direction) % options.length;
+
+    options[nextElementPosition].focus();
+  }
+
+  const onKeyDownHandler = (e: React.KeyboardEvent<HTMLDivElement>) => {
+    e.stopPropagation();
+    switch (e.key) {
+    case Keys.ARROW_DOWN :
+      focusNextOption(e.currentTarget, 1);
+      break;
+    case Keys.ARROW_UP:
+      focusNextOption(e.currentTarget, -1);
+      break;
+    case Keys.ENTER:
+      onActivate(e);
+    }
+  }
+
   return (
-    <div {...rest} className={classes} onClick={onClick}>
+    <div {...rest} className={classes} onClick={onClick || onActivate} ref={forwardRef} onKeyDown={onKeyDownHandler} tabIndex={-1}>
```

Comment:
I think that we should keep binding onClick={onClick}

Reply:
Since both clicking and pressing enter do the same thing, ideally we only want one handler "onActivate", doesn't matter whether the user clicked or pressed enter. So ideally we want `onClick={onActivate}` as well. I just kept the onClick for backward compatibility. In the future I think you can pass just `onActivate` and it will be called no matter you click or press enter.

Reply:
The reason why I did not change the type of onClick was that onClick sounds like a mouse event so it might be confusing to add the KeyboardEvent type to it. But I can change it if that makes the code simpler.